frontend/src/v5/ui/routes/userSignup/userSignupForm/userSignupFormStep/userSignupFormStepAccount/userSignupFormStepAccount.component.tsx

```diff
@@ -0,0 +1,192 @@
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { formatMessage } from '@/v5/services/intl';
+import { useEffect } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useForm } from 'react-hook-form';
+import { FormTextField } from '@controls/formTextField/formTextField.component';
+import UserIcon from '@assets/icons/outlined/user-outlined.svg';
+import EmailIcon from '@assets/icons/outlined/email-outlined.svg';
+import PasswordIcon from '@assets/icons/outlined/lock-outlined.svg';
+import { UserSignupSchemaAccount } from '@/v5/validation/schemes';
+import { isEqual, pick, defaults } from 'lodash';
+import { NextStepButton } from '../userSignupFormStep.styles';
+import { IconContainer } from './userSignupFormStepAccount.styles';
+
+export interface IAccountFormInput {
+	username: string;
+	email: string;
+	password: string;
+	confirmPassword: string;
+}
+
+type UserSignupFormStepAccountProps = {
+	updateFields: (fields: any) => void;
+	onSubmitStep: () => void;
+	onComplete: () => void;
+	onUncomplete: () => void;
+	fields: IAccountFormInput;
+	alreadyExistingUsernames: string[];
+	alreadyExistingEmails: string[];
+};
+
+export const UserSignupFormStepAccount = ({
+	updateFields,
+	onSubmitStep,
+	onComplete,
+	onUncomplete,
+	fields,
+	alreadyExistingUsernames,
+	alreadyExistingEmails,
+}: UserSignupFormStepAccountProps) => {
+	const DEFAULT_FIELDS: IAccountFormInput = {
+		username: '',
+		email: '',
+		password: '',
+		confirmPassword: '',
+	};
+
+	const getAccountFields = (): IAccountFormInput => defaults(
+		pick(fields, ['username', 'email', 'password', 'confirmPassword']),
+		DEFAULT_FIELDS,
+	);
+
+	const {
+		watch,
+		getValues,
+		trigger,
+		control,
+		formState,
+		formState: { errors, isValid: formIsValid, dirtyFields },
+	} = useForm<IAccountFormInput>({
+		mode: 'all',
+		reValidateMode: 'onChange',
+		resolver: yupResolver(UserSignupSchemaAccount(alreadyExistingUsernames, alreadyExistingEmails)),
+		defaultValues: getAccountFields(),
+	});
+
+	const password = watch('password');
+
+	useEffect(() => {
+		if (dirtyFields.password && dirtyFields.confirmPassword) {
+			trigger('confirmPassword');
+		}
+	}, [password]);
+
+	useEffect(() => {
+		if (formIsValid) {
+			onComplete();
+		} else {
+			onUncomplete();
+			if (alreadyExistingUsernames.length) trigger('username');
+			if (alreadyExistingEmails.length) trigger('email');
```

Comment:
I think there's a way to async validate the fields.
https://react-hook-form.com/advanced-usage 
I don't know if this could help. If it leads to worst code, disregard this comment (well, that should always be the case 😁 )

Reply:
The problem is due to the rendering of the component. Since the form is using a stepper, the `AccountStep` is destroyed before reaching the submit step. When the user submits the form, if either the username or the email already exist, the user is brought back to the `AccountStep`, but the latter needs to be re-rendered. RHF does not automatically triggers the validation at this point, so the `usernname/email is already taken` error does not show up.